### PR TITLE
[Grid.Cell] Add column, and row props

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes #0000 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Grid/components/Cell/Cell.scss [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  --pc-row-xs: initial;
+  --pc-row-sm: var(--pc-row-xs);
+  --pc-row-md: var(--pc-row-sm);
+  --pc-row-lg: var(--pc-row-md);
+  --pc-row-xl: var(--pc-row-lg);
+  --pc-column-xs: initial;
+  --pc-column-sm: var(--pc-column-xs);
+  --pc-column-md: var(--pc-column-sm);
+  --pc-column-lg: var(--pc-column-md);
+  --pc-column-xl: var(--pc-column-lg);
```
Comment (alex-page) [inline polaris-react/src/components/Grid/components/Cell/Cell.scss:14]: This code is pretty confusing. Are we setting everything to `initial`? I don't see a lot of values in the `--pc-` approach here.

Comment (kyledurand) [reply polaris-react/src/components/Grid/components/Cell/Cell.scss:14]: Yeah it's hard to understand at first but I think it's better than doing [this](https://github.com/Shopify/polaris/pull/5966/commits/e6d59c002e75e544f36262bbbc6e416905473dfa#diff-44cc555ece4f1fd992b5d469080859d4487ca98bfe7e1d1b95283ec57340cc05R49-R54) even though that might be easier to understand?

It was @aaronccasanova 's suggestion and I liked it better than calling var() a ton of times in what is essentially just as hard to read css

Comment (kyledurand) [reply polaris-react/src/components/Grid/components/Cell/Cell.scss:14]: This is so you only have to set things like row, column, grid-template-areas at `xs and lg` as a minimum vs having to set them all. xs = 6 column, lg is where we switch to 12 and they cascade from a mobile up perspective

Comment (aaronccasanova) [reply polaris-react/src/components/Grid/components/Cell/Cell.scss:14]: @alex-page See this comment and associated demo of the behavior:
https://github.com/Shopify/polaris/pull/5966#discussion_r888569425

As Kyle mentioned, this enables the mobile first props strategy used throughout the component.
Before:
```jsx
<Grid columns={{xs: 4, sm: 4, md: 4, lg: 6, xl: 6}} />
```
After:
```jsx
<Grid columns={{xs: 4, lg: 6}} />
```

Comment (alex-page) [reply polaris-react/src/components/Grid/components/Cell/Cell.scss:14]: Could we leave a little comment above this css that explains what it is doing. It makes sense to me but took a while to understand what is going on.

**Review by alex-page — APPROVED** [review]

File: polaris-react/src/components/Grid/components/Cell/Cell.tsx [hunk 2]
```diff
@@ -18,22 +25,44 @@ interface Columns {
...
-    columnSpan?.xs && styles[`grid-${columnSpan?.xs}-column-xs`],
-    columnSpan?.sm && styles[`grid-${columnSpan?.sm}-column-sm`],
-    columnSpan?.md && styles[`grid-${columnSpan?.md}-column-md`],
-    columnSpan?.lg && styles[`grid-${columnSpan?.lg}-column-lg`],
-    columnSpan?.xl && styles[`grid-${columnSpan?.xl}-column-xl`],
+    columnSpan?.xs && styles[`Cell-${columnSpan?.xs}-column-xs`],
+    columnSpan?.sm && styles[`Cell-${columnSpan?.sm}-column-sm`],
+    columnSpan?.md && styles[`Cell-${columnSpan?.md}-column-md`],
+    columnSpan?.lg && styles[`Cell-${columnSpan?.lg}-column-lg`],
+    columnSpan?.xl && styles[`Cell-${columnSpan?.xl}-column-xl`],
```
Comment (aaronccasanova) [inline polaris-react/src/components/Grid/components/Cell/Cell.tsx]:
```suggestion
    columnSpan?.xs && styles[`Cell-${columnSpan.xs}-column-xs`],
    columnSpan?.sm && styles[`Cell-${columnSpan.sm}-column-sm`],
    columnSpan?.md && styles[`Cell-${columnSpan.md}-column-md`],
    columnSpan?.lg && styles[`Cell-${columnSpan.lg}-column-lg`],
    columnSpan?.xl && styles[`Cell-${columnSpan.xl}-column-xl`],
```

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**